### DISCUSSION TITLE
Endpoint Routes - add support for regular expressions, evaluation order for routes

### DISCUSSION THREAD

**Discussion by eric-anderton-at-rearc** [noop-inc/documentation#167] [Feature Requests and Ideas]
Noop does a good job of providing redirects and routing for static rules, AKA, Endpoint Routes.  There are two features that put this behind the capabilities of Nginx, Apache, CloudFront, etc.  That's matching of paths via regular expressions, and specifying an order of evaluation for multiple routes.

I think the Endpoint Routes can be made sortable with drag controls and maybe arrows to shuttle routes up and down the list.  This would create a priority order of evaluation, so catch-all rules can be placed at the bottom, and more specific rules at the top.

Regex support could be made available with a "regex" or "PCRE" checkbox after "All Routes" is unchecked.  It would also be nice to have a checkbox for case-insensitive matching.

Alternately, regex matching via Noop YAML logic would be a way to implement this.  The `glob` operator exists, but can't touch a range a problems that regular expressions can.

**Comment by derekwolpert** [comment]
@eric-anderton-at-rearc

> _I think the Endpoint Routes can be made sortable with drag controls and maybe arrows to shuttle routes up and down the list. This would create a priority order of evaluation, so catch-all rules can be placed at the bottom, and more specific rules at the top._

Endpoint Routes actually already behave exactly as you are desiring. When a request is processed Endpoint Routes are evaluated in the order they are defined (first to last), so you should absolutely order routes based on priority, catch-all considerations, etc. 

Additionally the Endpoint Routes form supports draggable inputs for reordering/sorting updates. See the demo video below for further context (thanks for the feedback here, as it highlights this drag functionality might need to be communicated more clearly within the UI).

https://github.com/noop-inc/documentation/assets/48101033/de8197e9-806c-43da-8910-b5fd078f50ec

> _Regex support could be made available with a "regex" or "PCRE" checkbox after "All Routes" is unchecked. It would also be nice to have a checkbox for case-insensitive matching._
> 
> _Alternately, regex matching via Noop YAML logic would be a way to implement this. The glob operator exists, but can't touch a range a problems that regular expressions can._

Definitely valid feedback here. We do intend to add a `regex` operator to Noop Logic in the future, which would be usable as an alternative means to pattern match pathnames when Endpoint Routes are evaluated. We'll be sure to give you a heads up when this feature is live.

**Reply by eric-anderton-at-rearc** [reply]
@derekwolpert - Thank you.  I don't know how I missed the DnD support here.  Thank you for taking the time to explain it all in such detail.  I appreciate it.
